src/courses/mappers/06.md

```diff
@@ -0,0 +1,309 @@
+---
+order: 6
+next: 07.md
+title: Formal OHDF Schema Definition
+author: Charles Hu
+---
+
+## Breaking Down the Formal OHDF Schema
+
+We can break down the formal OHDF schema in the same manner as before by observing each structure in the hierarchy from a top-down view.
+
+::: note Full OHDF Schema
+The following section contains a breakdown of a streamlined version of the OHDF schema. For the full and up-to-date version, see [here](https://saf.mitre.org/framework/normalize/ohdf-schema).
+:::
+
+### 1. <i>**Profiles**</i>
+
+::: details Profiles Schema
+```json
+profiles: [
+  0: {
+    name,
+    version,
+    sha256,
+    title,
+    maintainer,
+    summary,
+    license,
+    copyright,
+    copyright_email,
+    supports,
+    attributes,
+    groups,
+    controls,
+    status
+  },
+  ... // More items may exist if the security service produces multiple profiles per export
+]
+```
+:::
+
+- `name`: Name of profile, usually the original security service tool; should be unique.
+- `version`: Version of security service tool.
+- `sha256`: Checksum of the profile (NOTE: AUTOMATICALLY GENERATED BY HDF CONVERTERS, DO NOT POPULATE).
+- `title`: Title of security service scan; should be human readable.
+- `maintainer`: Maintainer.
+- `summary`: Summary of security service export (e.g., the STIG header).
+- `license`: Copyright license.
+- `copyright`: Copyright holder.
+- `copyright_email`: Copyright holder's email.
+- `supports`: Supported platform targets.
+- `attributes`: Inputs/attributes used in scan.
+- `groups`: Set of descriptions for the control groups (e.g., control IDs).
+- `controls`: Controls substructure (see below).
+- `status`: Status of profile (typically 'loaded').
+
+#### Example
+
+Here's an example of a <i>profiles</i> section from a converted GoSec data file:
+
+::: details OHDF Converted GoSec Data
+```json
+"profiles": [
```

Comment:
actually thinking about it a bit more, this is a good opportunity to have student interaction if we have a table as i described elsewhere.  my idea is that you actually showcase a snippet of the raw gosec results and then ask the student what info from here would fit the profile attributes section.  i get that the deeper mapper explanation stuff is going to be coming up later on.  this is intentionally gonna be super bare boned just to get the student into the mindframe of looking at random results files and thinking how it should look in hdf.  have the 'answer' to the question be "well here's what we came up with ourselves from the actual mapper".

Reply:
repeat the above for each of the execjson / profiles / controls / results levels.

Reply:
Addressed.